#!/usr/bin/env python
PACKAGE = "delta_2"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("disable_movement",   bool_t,   0, "prevent setpoints from being pinged to dynamixels",  False)
gen.add("read_positions",   bool_t,   0, "enable reading and publishing of servo positions",  True)
gen.add("vel_limit",    int_t, 0, "Servo velocity limit in bits",   0,   0,      1023)
gen.add("P",            int_t, 0, "Servo P gain",           1000,    0,      2000)
gen.add("I",            int_t, 0, "Servo I gain",           0,      0,      2000)
gen.add("D",            int_t, 0, "Servo D gain",           0,      0,      2000)


exit(gen.generate(PACKAGE, "delta_2", "Servo"))